config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -213,6 +218,11 @@ public JwtConfigurer jwkSetUri(String uri) {
 			return OAuth2ResourceServerConfigurer.this;
 		}
 
+		public JwtConfigurer publicKey(RSAPublicKey key) {
```

Comment:
I'm wondering if `verifyKey` make more sense. We need to consider when we support JWE and what that method name will be. The current name `publicKey` will collide between verification and decryption.

Reply:
Actually, I think that publicKey would never collide so long as we always take public keys (private keys for decryption).

But, you are right because of symmetric keys, which I wasn't thinking about until you mentionoed this. Eventually, I would suppose that this method will take just `javax.security.Key`, which would actually then not only collide, but it would also be semantically untrue since I'm certainly *not* giving it a public key.

Reply:
You are right, it wouldn't collide since `PrivateKey` is used for decryption.

I still feel `verifyKey` is more intuitive than `publicKey`.

I'll put my proposal here and we can gather feedback from @rwinch.

```
verifyKey(java.security.interfaces.RSAPublicKey key)		// Verify signature

decryptKey(java.security.interfaces.RSAPrivateKey key)	// Decrypt content

sharedKey(javax.crypto.SecretKey key)	// Encrypt/Decrypt content
```

Reply:
To me, invoking `decryptKey` indicates that you intend to perform (or allow) decryption in this app.

If I am using a shared key and only intend to verify signatures, how would I do this in your proposal? e.g. I don't want to "enable" decryption just because I'm using a shared key.

Reply:
At least for now, I would remove this all together. This is not a very common case, so users can just either expose a JwtDecoder bean or inject the JwtDecoder into the DSL.